Source/Core/Common/FormatUtil.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstddef>
+#include <string_view>
+
+namespace Common
+{
+constexpr std::size_t CountFmtReplacementFields(std::string_view s)
+{
+  std::size_t count = 0;
+  for (std::size_t i = 0; i < s.size(); ++i)
+  {
+    if (s[i] != '{')
+      continue;
+
+    // If the opening brace is followed by another brace, what we have is
+    // an escaped brace, not a replacement field.
+    if (i + 1 < s.size() && s[i + 1] == '{')
```

Comment:
I might be mixing up projects, but don't we have a string somewhere that attempts to output `{something}` and looks like `{{{}}}` in the string? The first two `{{` for a literal `{` and the third `{` as the inner `fmt` placeholder?

This might not work correctly with this validation.

Reply:
I think you're right, I also seem to remember that we have such a string so I made it a test case when I was testing this on godbolt: https://godbolt.org/z/GnKvWz

It should work fine because when an escaped brace is detected, the second brace character is skipped, so the third brace character isn't detected as being part of an `{{` escape sequence for the second character. 

As you can see in the first and third examples, `{{{}}}` actually works fine

Reply:
I added this explanation to the comment and some static_assert tests